packages/compass-collection-stats/src/stores/store.js

```diff
@@ -1,270 +1,148 @@
-        totalIndexSize: this._format(result.index_size, 'b'),
-        avgIndexSize: this._format(this._avg(result.index_size, result.index_count), 'b'),
-        rawDocumentCount: result.document_count,
-        rawTotalDocumentSize: result.document_size,
-        rawAvgDocumentSize: this._avg(result.document_size, result.document_count),
-        rawIndexCount: result.index_count,
-        rawTotalIndexSize: result.index_size,
-        rawAvgIndexSize: this._avg(result.index_size, result.index_count)
-      };
-    },
-
-    _avg(size, count) {
-      if (count <= 0) {
-        return 0;
-      }
-      return size / count;
-    },
-
-    _format(value, format = 'a') {
-      const precision = value <= 1000 ? '0' : '0.0';
-      return numeral(value).format(precision + format);
-    }
-  });
-
-  // Set the app registry if preset. This must happen first.
-  if (options.localAppRegistry) {
-    const localAppRegistry = options.localAppRegistry;
-    setLocalAppRegistry(store, localAppRegistry);
-
-    /**
-     * When the collection is changed, update the store.
-     */
-    localAppRegistry.on('import-finished', () => {
-      refreshInput(store);
-    });
-
-    /**
-     * Refresh documents on data refresh.
-     */
-    localAppRegistry.on('refresh-data', () => {
-      refreshInput(store);
-    });
-
-    /**
-     * Refresh documents on document deletion.
-     */
-    localAppRegistry.on('document-deleted', () => {
-      refreshInput(store);
-    });
-
-    /**
-     * Refresh documents on document insertion.
-     */
-    localAppRegistry.on('document-inserted', () => {
-      refreshInput(store);
-    });
+const configureStore = ({ namespace, globalAppRegistry } = {}) => {
+  const { instance } =
+    globalAppRegistry.getStore('App.InstanceStore')?.getState() ?? {};
+
+  // We only want to attach listeners to the instance once if we haven't done it
+  // before, we use stored instances as a flag to determine if we did it already
+  if (!store[kGlobalAppRegistry]) {
+    store[kGlobalAppRegistry] = globalAppRegistry;
+    globalAppRegistry.on('instance-destroyed', onInstanceDestroyed);
   }
 
-  if (options.globalAppRegistry) {
-    const globalAppRegistry = options.globalAppRegistry;
-    setGlobalAppRegistry(store, globalAppRegistry);
+  if (!store[kInstance]) {
+    store[kInstance] = instance;
+    instance.on('change:collections.status', onCollectionStatusChange);
   }
 
-  // Set the data provider - this must happen second.
-  if (options.dataProvider) {
-    setDataProvider(
-      store,
-      options.dataProvider.error,
-      options.dataProvider.dataProvider
-    );
-  }
+  store.setState({ namespace });
 
-  if (options.isReadonly) {
-    setIsReadonly(store, options.isReadonly);
-  }
+  const { database, ns } = toNS(namespace);
 
-  if (options.isTimeSeries) {
-    setIsTimeSeries(store, options.isTimeSeries);
-  }
+  const coll = instance.databases.get(database)?.collections.get(ns) ?? {};
```

Comment:
Does it make sense to `store.updateCollectionDetails({})`? How would we get into that kind of situation and do we support it?

Reply:
We probably can't in any flow we can predict, but FWIW I don't think that it's inherently bad Compass doesn't end any weird condition it can run into with a hard crash, we are connecting it to basically whatever users want and don't really control the backend our app communicates with, so making code continue to run in some cases where it's not that critical doesn't really bother me. I am happy and open to get another perspective on that though

Reply:
I guess I'm just very averse to code that never executes and should never execute. Especially if we don't test or otherwise support those cases. I worry that it will just hide the real problems and cause hard to debug problems elsewhere. Generally I prefer to have things throw so we can at least find out about it. But I don't have enough context to know if this is just another case of the event based architecture I pointed out above - ie maybe we _can_ hit these conditions because other code is already written to be this forgiving and just continue rather than either be architected in such a way that these conditions aren't possible or at least throwing if the impossible happens.

Defensive/forgiving coding tends to cause a cascade of more defensive/forgiving coding elsewhere.

But that's just my opinion - feel free to ignore.

Reply:
Well I asked for the opinion, so I'm definitely not just ignoring it! 😄 I generally agree and as we touched on in a few other places it's just hard to do right now due to how Compass works, I will definitely pay more attention to this now and re-evaluate if we can handle some cases better in that sense, thanks for sharing your thoughts on this!

Reply:
My hope is that if we ever change this code to typescript it will force us to at least deal with some of these.